src/ucp/wireup/wireup.c

```diff
@@ -1525,7 +1593,10 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key,
                     "new_key->wireup_msg_lane=%u", new_key->wireup_msg_lane);
     }
 
-    wireup_lane = ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST);
+    is_am_replaced = ucp_wireup_is_am_lane_replaced(ep, reuse_lane_map);
+    wireup_lane    = is_am_replaced ?
+                                ucp_ep_get_am_lane(ep) :
+                                ucp_wireup_get_msg_lane(ep, UCP_WIREUP_MSG_REQUEST);
```

Comment:
th elogic is not clear, and why only for `UCP_WIREUP_MSG_REQUEST`?
maybe add new function `ucp_lane_index_t ucp_config_key_get_msg_lane(const ucp_ep_config_t *ep_config)` and call it from `ucp_wireup_get_msg_lane` and here:
```
wireup_lane = ucp_config_key_get_msg_lane(new_key);
```

Reply:
not sure how can it be done, can you share the contents of such function?

Reply:
based on code of `ucp_wireup_get_msg_lane`:
```
    if (msg_type == UCP_WIREUP_MSG_ACK) {
        lane          = ep_config->key.am_lane;
        fallback_lane = ep_config->key.wireup_msg_lane;
    } else {
        /* for request/response, try wireup_msg_lane first */
        lane          = ep_config->key.wireup_msg_lane;
        fallback_lane = ep_config->key.am_lane;
    }
```
> why only for UCP_WIREUP_MSG_REQUEST?

just to go via `else` branch?